ffmpeg/ffmpeg.go

```diff
@@ -490,7 +505,81 @@ func Transcode3(input *TranscodeOptionsIn, ps []TranscodeOptions) (*TranscodeRes
 	return t.Transcode(input, ps)
 }
 
+type CodingSizeLimit struct {
+	WidthMin, HeightMin int
+	WidthMax, HeightMax int
+}
+
+func clamp(val, min, max int) int {
+	if val <= min {
+		return min
+	}
+	if val >= max {
+		return max
+	}
+	return val
+}
+
+func (l *CodingSizeLimit) Clamp(p *VideoProfile, format MediaFormatInfo) error {
+	w, h, err := VideoProfileResolution(*p)
+	if err != nil {
+		return err
+	}
+	// detect correct rotation
+	outputAr := float32(w) / float32(h)
+	inputAr := float32(format.Width) / float32(format.Height)
+	if (inputAr > 1.0) != (outputAr > 1.0) {
+		// comparing landscape to portrait, apply rotate on chosen resolution
+		w, h = h, w
+	}
+	// Adjust to minimal encode dimensions keeping aspect ratio
+	landscapeW := clamp(w, l.WidthMin, l.WidthMax)
+	landscapeH := format.ScaledHeight(landscapeW)
+	portraitH := clamp(h, l.HeightMin, l.HeightMax)
+	portraitW := format.ScaledWidth(portraitH)
+	// Choose larger option
+	if landscapeW*landscapeH > portraitH*portraitW {
+		p.Resolution = fmt.Sprintf("%dx%d", landscapeW, landscapeH)
```

Comment:
Is `p.Resolution` rotation-sensitive?

I guess that because of the line 533 (`w, h = h, w`) you may sometimes end up using portrait or landscape here.

Reply:
Yes resolution is rotation sensitive, that is why line 533 converts landscape to portrait if source video is in portrait.

The landscape and portrait are misleading here 🤦‍♂️ let me fix this